### PR TITLE
Initial attempt at fixing integration-tests.

### LINKED ISSUE #11565: CircleCI builds are failing
Moving this to addons-frontend, now that legacy frontend is gone is another thing but we should put circleci tests to green in the meantime.

```
tests/ui/test_home.py::test_category_section_loads_correct_category[Resolution: 414x738] FAILED
```

And it seems to fail because `.CategoryHeader-name` isn't present on this page: 

index

Which links to this commit on addons-frontend: https://github.com/mozilla/addons-frontend/commit/f7d85d5d88233d60cb19726c57463f13da14f0c0

### PR DESCRIPTION
**Pull request by jrbenny35**
This hopefully fixes the integration tests before the BIG SPLIT. 
Fixes #11565 #11720 #11721
Works on #11691

### CODE REVIEW

**Comment by EnTeQuAk** [comment]
You may have to take a look at the code-style related failures in travis: https://travis-ci.org/mozilla/addons-server/jobs/556537194#L515-L536

The circleci issue doesn't look permanent but you may have a much better idea about what's going on there.

File: src/olympia/landfill/serializers.py [hunk 1]
```diff
@@ -103,25 +130,36 @@ def create_named_addon_with_author(self, name, author=None):
         if author is None:
             author = user_factory()
         try:
-            generate_user(author)
+            user = UserProfile.objects.create(username=author, email=f'{author}@email.com')
+            user.id = 4757633
```
Comment (EnTeQuAk) [inline src/olympia/landfill/serializers.py]: You will have to use `user.update(id=settings.4757633)` ideally to not hardcode the id.

File: src/olympia/landfill/serializers.py [hunk 2]
```diff
@@ -103,25 +130,36 @@ def create_named_addon_with_author(self, name, author=None):
...
                 name=u'{}'.format(name),
                 slug=u'{}'.format(name),
+                version_kw={
+                'recommendation_approved': True,
+                'nomination': days_ago(6)
+                }
             )
             addon.save()
         else:
+            author.id = 4757633
```
Comment (EnTeQuAk) [inline src/olympia/landfill/serializers.py]: Same here, you will have to use author.update(id=settings.TASK_USER_ID)` ideally to not hardcode the id.

File: src/olympia/landfill/serializers.py [hunk 3]
```diff
@@ -103,25 +130,36 @@ def create_named_addon_with_author(self, name, author=None):
...
             # If the user is already made, use that same user,
             # if not use created user
             addon = addon_factory(
-                status=STATUS_APPROVED,
+                status=amo.STATUS_APPROVED,
                 users=[UserProfile.objects.get(username=author)],
                 name=u'{}'.format(name),
                 slug=u'{}'.format(name),
+                version_kw={
+                'recommendation_approved': True,
```
Comment (EnTeQuAk) [inline src/olympia/landfill/serializers.py]: Look at the codestyle issues in travis, this needs indentation.

File: tests/ui/pages/desktop/base.py [hunk 4]
```diff
@@ -11,7 +11,7 @@ class Base(Page):
 
     def __init__(self, selenium, base_url, locale='en-US', **kwargs):
         super(Base, self).__init__(
-            selenium, base_url, locale=locale, timeout=30, **kwargs)
+            selenium, base_url, locale=locale, timeout=10, **kwargs)
```
Comment (EnTeQuAk) [inline tests/ui/pages/desktop/base.py]: Is that a performance optimization or why did we change that?

File: tests/ui/pages/desktop/devhub.py [hunk 5]
```diff
@@ -142,12 +142,14 @@ class Header(Region):
         _sign_out_locator = (By.CSS_SELECTOR, '.DevHub-Navigation-SignOut > a')
 
         def click_login(self):
+            self.selenium.find_element_by_id("djHideToolBarButton").click()
```
Comment (EnTeQuAk) [inline tests/ui/pages/desktop/devhub.py]: Is that actually needed, does it avoid the other buttons to be clicked? That'd be funny

File: tests/ui/pages/desktop/devhub.py [hunk 6]
```diff
@@ -133,7 +133,7 @@ class Header(Region):
 
         _register_link_locator = (
             By.CSS_SELECTOR,
-            '.DevHub-Navigation-Register > a:nth-child(1)'
+            'li.show-on-desktop:nth-child(7) > a:nth-child(1)'
```
Comment (EnTeQuAk) [inline tests/ui/pages/desktop/devhub.py]: Why is that necessary? The class `DevHub-Navigation-Register` still exists, these pages haven't changed since forever.

File: tests/ui/test_search.py [hunk 7]
```diff
@@ -127,6 +120,7 @@ def test_long_terms_dont_break_suggestions(base_url, selenium):
 
 
 @pytest.mark.nondestructive
+@pytest.mark.skip(reason='Search Behavior has changed')
```
Comment (EnTeQuAk) [inline tests/ui/test_search.py]: Let's create an issue if that leads to skipping the test.

File: tests/ui/pages/desktop/base.py (hunk 4 above)
Comment (jrbenny35) [reply tests/ui/pages/desktop/base.py]: This can be removed

File: tests/ui/pages/desktop/devhub.py (hunk 5 above)
Comment (jrbenny35) [reply tests/ui/pages/desktop/devhub.py]: Yeah it is somehow blocking the other buttons.

File: tests/ui/pages/desktop/devhub.py (hunk 6 above)
Comment (jrbenny35) [reply tests/ui/pages/desktop/devhub.py]: I think this was a change due to the django toolbar. It should be able to be reverted.

File: tests/ui/test_search.py (hunk 7 above)
Comment (jrbenny35) [reply tests/ui/test_search.py]: Trying this again with proper recommendations seems to work fine, so I'll "unskip" it.

**Comment by jrbenny35** [comment]
The circleci failure is an error from what seems to be the devhub "Submit a new add-on" page. Doesn't seem like the checkboxes are clickable. I cant reproduce it locally though..

File: tests/ui/pages/desktop/devhub.py (hunk 5 above)
Comment (EnTeQuAk) [reply tests/ui/pages/desktop/devhub.py]: Probably worth adding a comment. Also, in case we disable debug toolbar by default locally in the future, does that work then too or does the "click" fail loudly?

Comment (jrbenny35) [reply tests/ui/pages/desktop/devhub.py]: It will fail to find the element, and that would be loud.

Comment (EnTeQuAk) [reply tests/ui/pages/desktop/devhub.py]: Should be fine I guess :man_shrugging:

**Comment by EnTeQuAk** [comment]
> This hopefully fixes the integration tests before the BIG SPLIT.
Fixes #11565 #11691 #11720 #11721

It prepares for  #11691, it doesn't really fix it though
